### PR TITLE
Change setup of parcel correction tests

### PR DESCRIPTION
**Pull request by matt-frey**

### CODE REVIEW

File: unit-tests/2d/test_parcel_correction.f90 [hunk 1]
```diff
@@ -77,14 +60,50 @@ program test_parcel_correction
...
+
+        tmp = m * val + q
+        parcels%position(1, n) = parcels%position(1, n) + tmp
+
+        call random_number(val)
+        tmp = m * val + q
+        parcels%position(2, n) = parcels%position(2, n) + tmp
+
+        call apply_periodic_bc(parcels%position(:, n))
+        call apply_reflective_bc(parcels%position(:, n), parcels%B(:, n))
```
Comment (sjboeing) [inline unit-tests/2d/test_parcel_correction.f90:87]: Note that this initialisation (with reflective BC) may mean that parcels are less likely to be near a boundary as compared to inside the interior. Not sure if this is the case, would need to give it some thought, or did you think about this already?

Comment (sjboeing) [reply unit-tests/2d/test_parcel_correction.f90:87]: It is not a very important issue in any case.

Comment (matt-frey) [reply unit-tests/2d/test_parcel_correction.f90:87]: I try to avoid parcel centres being outside the box.

Comment (matt-frey) [reply unit-tests/2d/test_parcel_correction.f90:87]: Parcels can still be near the boundary, but they should not be outside. Parcels are only "mirrored" when they are outside the specified domain. If they are just a little bit outside they will end up being just a little bit inside, so close to the boundary.

Comment (sjboeing) [reply unit-tests/2d/test_parcel_correction.f90:87]: Ah yes, can see this works out when n_per_dim=2. Happy for you to merge.

**Review by sjboeing — APPROVED** [review]

**Pull request merged**